climada/util/yearsets.py

```diff
@@ -0,0 +1,244 @@
+                                 of given sampled_years)
+      by sampling events from the existing input event_impacts with a Poisson distribution
+      centered around n_events per year (n_events = sum(event_impacts.frequency)).
+
+    Parameters:
+      event_impacts : impact object
+          impact object containing impacts per event
+    Optional parameters:
+        sampled_years : int or list
+            Either an integer specifying the amount of years to
+            be sampled or a list of years that shall be covered by the resulting annual_impacts
+            default: a 1000 year-long list starting in the year 0001
+        sampling_vect : dict
+            The sampling vector specifying which events contained in the events_impacts are
+            selected in the newly created annual_impacts and the number of events per year
+            in the annual_impacts, therefore the sampling_vect contains two arrays:
+                selected_events (array): sampled events (len: total amount of sampled events)
+                events_per_year (array): events per sampled year
+            The sampling_vector needs to be obtained in a first call,
+            i.e. [annual_impacts, sampling_vect] = climada_yearsets.impact_yearset(...)
+            and can then be provided in subsequent calls(s) to obtain the exact same sampling
+            (also for a different event_impacts object)
+        correction_fac : boolean
+            If True the resulting annual_impacts are scaled in so that the expected annual
+            impact (eai) of the annual_impacts = eai of the events_impacts
+
+    Returns:
+      annual_impacts : impact object
+          annual impacts for all sampled_years
+      sampling_vect : dict
+          the sampling vector containing two arrays:
+              selected_events (array) : sampled events (len: total amount of sampled events)
+              events_per_year (array) : events per sampled year
+          Can be used to re-create the exact same annual_impacts yearset
+      """
+
+
+    if not sampled_years and not sampling_vect:
+        sampled_years = np.arange(1, 1000+1).tolist()
+    elif not sampled_years:
+        sampled_years = np.arange(1, len(sampling_vect['selected_events'])+1).tolist()
+    elif isinstance(sampled_years, int):
+        sampled_years = np.arange(1, sampled_years+1).tolist()
+
+    if sampling_vect:
+        if len(sampled_years) != len(sampling_vect['events_per_year']):
+            raise ValueError("The number of sampled_years and the length of the list of "
+                             "events_per_year in the sampling vector must be equal.")
+        if sum(sampling_vect['events_per_year']) != len(sampling_vect['selected_events']):
+            raise ValueError("The sampling vector is faulty: the sum of selected events "
+                             "does not correspond to the number of selected events.")
+
+    year_list = [str(date) + '-01-01' for date in sampled_years]
```

Comment:
This is never actually used right? Then it is probably better to remove it here and directly set
```
n_sampled_years = len(sampled_years)
annual_impacts.date = u_dt.str_to_date([str(year) + '-01-01' for year in sampled_years])
```